### ISSUE TITLE
Arch User Repository Package for experimental py3 pwntools

### ISSUE THREAD

**Issue by rnbwdsh** [Gallopsled/pwntools#1220]
As Arch Linux often runs into issues when installing from pip instead of the package manager (pacman), it'd be awesome if [the pwntools 3 package from pypi](https://pypi.org/project/pwntools/) also was available in AUR, not just [the stable python2](https://aur.archlinux.org/packages/python2-pwntools/) implementation.

If that seems to be too much work, [there even seems to be a script](https://github.com/lclarkmichalek/pip2arch) for that, or I could help.

Otherwise, can i create a git-based AUR package based on [the python3 v2.2 fork](https://github.com/arthaud/python3-pwntools), like in this [PKGBUILD](https://github.com/Gallopsled/pwntools/files/2526303/PKGBUILD.txt) file?

And yes, I'm aware the py3 branch is very experimental and TTY doesn't work and the #529 issue, but I think a 90% working version is fine for a rolling-release distro that's updated quite often, because patches reach the users.

**Comment by zachriggle** [comment]
The "pwntools 3 package from pypi" is the Python2 version.  There have never been a `python3` release.

The old `python3` fork from v2.2 was never completed, nor passed tests -- it's also entirely unsupported / dead / EOL / will not get bugfixes, ever.

All that said, you can create whatever you'd like on AUR but please do not name it `pwntools`, as this could be confusing and people will try to file bug reports here.  At a minimum, prefix it with something else, e.g. `rnbdsh-pwntools-python3`.

Separately, we'd love contributions to Pwntools which move toward Python3 support (see e.g. #1176)

**Comment by rnbwdsh** [comment]
>The pwntools 3 package from pypi" is the Python2 version. There have never been a python3 release.
Oh, currently pip3 install pwntools works (as the python_require isn't set in stable) - as the     python_requires      = '~=2.7', in the setup.py isn't backported to 3.12.
Sidestory: checking out 3.14 (dev) and running setup.py doesn't work due to [this error](https://pastebin.com/3WYX8gUW). Patching out the rfc822_escape in `/usr/lib/python3.7/site-packages/setuptools/dist.py` resolves this. The cause seems that the self-object.description contains bytes, but wants to split by a string.

>The old python3 fork from v2.2 was never completed, nor passed tests -- it's also entirely unsupported / dead / EOL / will not get bugfixes, ever.
Even though it took me quite some time / dockering to get the tests running at all under manjaro with a ubuntu18 docker, "only" 45 of the 1400 tests seem to fail -> quite ok.
Setting up the tests natively fails due to hardcoded apt-get and hardcoded binary paths (i.e. assuming "python" is python2)

>All that said, you can create whatever you'd like on AUR but please do not name it pwntools, as this could be confusing and people will try to file bug reports here. At a minimum, prefix it with something else, e.g. rnbdsh-pwntools-python3.
According to [AUR guidelines](https://wiki.archlinux.org/index.php/Python_package_guidelines), the naming scheme should be [python|python2]-pwntools[-...], so technically the official package doesn't follow the guidelines, and the conforming name for my py3-package would be python-pwntools2-legacy-git. (Which is kind of confusing because python-pwntools is a python2 package) -> suggestions?

>Separately, we'd love contributions to Pwntools which move toward Python3 support (see e.g. #1176)
That's kind of my next issue: In #529, it seems that there already exist kind-of-working, non-merged versions, so I'm not sure if creating a legacy-90%-test-passing package makes more sense than waiting for a current py3-compatibility merge. However, I'll look into making the test-environment at least pacman/arch-compatible.

Update: I just tested and checked out https://github.com/Arusekk/pwntools/tree/full-3-compat and it works like a breeze. Even though not all tests work for me (mostly due to bytes-to-string-comparison-asserts). I fixed some things in a fork of this branch.
